hyperapp.d.ts

```diff
@@ -17,10 +17,19 @@ export interface VNode<Attributes = {}> {
  *
  * @memberOf [VDOM]
  */
-export interface Component<Attributes = {}, State = {}, Actions = {}> {
-  (attributes: Attributes, children: Array<VNode | string>):
-    | VNode<Attributes>
-    | View<State, Actions>
+export interface Component<Attributes = {}> {
+  (attributes: Attributes, children: Array<VNode | string>): VNode<Attributes>
+}
+
+/** A LazyComponent is a Component that returns a view function.
+ *
+ * @memberOf [VDOM]
+ */
+export interface LazyComponent<Attributes = {}, State = {}, Actions = {}> {
```

Comment:
This shouldn't be a framework typing. It's not describing the API and is only for convenience. 

Also the naming is confusing. I think `ComponentCreator` would be a better name, since it's a function returning a component

Reply:
Hmm, the read me explicitly calls out Lazy Components as a distinct concept. Ie are passed state and actions not just attributes like non lazy components). Though it doesn't explicityly say that,

I just split it out the type from component typing as State and Actions are note part of a non lazy component so I thought is clearer.

It returns a View not a Component. And yes that is a bit confusing as a component returning a view seems weird to me given a view is something the App takes and manages. Still that's the way it is. :)

Reply:
Hmm, the read me explicitly calls out Lazt Components as a distinct concept (ie are passed state and actions not just attributes like non lazy components). I just split out the type from component.

It returns a View not a Component.

Reply:
I see it's a concept. I wasn't aware it's in the README. Still, it's not a description of a part of the API. Therefore I feel it shouldn't be in here.

/cc @jorgebucaran